include/boost/beast/core/buffer_traits.hpp

```diff
@@ -12,24 +12,62 @@
 
 #include <boost/beast/core/detail/config.hpp>
 #include <boost/asio/buffer.hpp>
+#include <boost/mp11/function.hpp>
 #include <type_traits>
 
 namespace boost {
 namespace beast {
 
-/** Type alias used to obtain the underlying buffer type of a buffer sequence.
+/** Determine if a list of types satisfy the <em>ConstBufferSequence</em> requirements.
+
+    This metafunction is used to determine if all of the specified types
+    meet the requirements for const buffer sequences. This type alias
+    will be `std::true_type` if each specified type meets the requirements,
+    otherwise, this type alias will be `std::false_type`.
+
+    @tparam TN A list of zero or more types to check. If this list is
+    empty, the resulting type alias will be `std::true_type`.
+*/
+#if BOOST_BEAST_DOXYGEN
+template<class... TN>
+struct is_const_buffer_sequence : __see_below__ {};
+//using is_const_buffer_sequence = __see_below__;
+#else
+template<class... TN>
+using is_const_buffer_sequence = mp11::mp_all<
+    net::is_const_buffer_sequence<TN>...>;
+#endif
+
+/** Determine if a list of types satisfy the <em>MutableBufferSequence</em> requirements.
+
+    This metafunction is used to determine if all of the specified types
+    meet the requirements for mutable buffer sequences. This type alias
```

Comment:
The documentation refers to "this type alias" whereas the user will see `struct ...`